src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +46,383 @@ ActorInit En_Takaraya_InitVars = {
+void func_80ADF654(EnTakaraya* this, PlayState* play) {
+    if (this->timer > 0) {
+        this->timer--;
+    } else {
+        if (gSaveContext.timerStates[TIMER_ID_MINIGAME_2] == TIMER_STATE_6) {
+            play->transitionType = TRANS_TYPE_80;
+            gSaveContext.nextTransitionType = TRANS_TYPE_FADE_WHITE;
+        } else {
+            play->transitionType = TRANS_TYPE_64;
+            gSaveContext.nextTransitionType = TRANS_TYPE_FADE_BLACK;
+        }
+        gSaveContext.nextCutsceneIndex = 0;
+        play->nextEntrance = ENTRANCE(TREASURE_CHEST_SHOP, 1);
+        play->transitionTrigger = TRANS_TRIGGER_START;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF050.s")
+void func_80ADF6DC(EnTakaraya* this) {
+    Animation_PlayLoop(&this->skelAnime, &object_bg_Anim_001384);
+    this->eyeTexIndex = 0;
+    this->actor.flags |= ACTOR_FLAG_10000;
+    this->actionFunc = func_80ADF730;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF2D4.s")
+void func_80ADF730(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actor.flags &= ~ACTOR_FLAG_10000;
+        func_80ADF7B8(this);
+    } else {
+        this->formSwitchFlag = gSaveContext.save.playerForm + this->switchFlag;
+        Actor_OfferTalk(&this->actor, play, 120.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF338.s")
+void func_80ADF7B8(EnTakaraya* this) {
+    this->actionFunc = func_80ADF7CC;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF4E0.s")
+void func_80ADF7CC(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF520.s")
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+        if (this->actor.textId == 0x77A) {
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED)) {
+                Message_CloseTextbox(play);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+                func_80ADEDF8(this);
+            } else {
+                this->actor.textId = 0x77C;
+                Message_ContinueTextbox(play, this->actor.textId);
+            }
+        } else {
+            this->actor.textId = D_80ADFB50[GET_PLAYER_FORM];
+            Message_ContinueTextbox(play, this->actor.textId);
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+            func_80ADF03C(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF608.s")
+void EnTakaraya_Update(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
+    s32 pad;
+    Vec3s torsoRot;
```

Comment:
Lame, she HAS torsoRot it was just put on a local where it couldn't be accessed in limb draws... Not sure if that is worth marking as a bug or not.

Reply:
Not really a bug imo. It wasn't used, so it was just a design decision to just use a stack variable. There are other instances of this same behavior elsewhere.